src/cudamatrix/cu-vector.cc

```diff
@@ -569,8 +569,13 @@ void CuVectorBase<Real>::AddDiagMat2(Real alpha, const CuMatrixBase<Real> &M,
   if (CuDevice::Instantiate().Enabled()) {
     if (dim_ == 0) return;
     MatrixTransposeType other_trans = (trans == kTrans ? kNoTrans : kTrans);
-    this->AddDiagMatMat(alpha, M, trans,
-                        M, other_trans, beta);
+    KALDI_ASSERT(dim_ == (trans == kNoTrans ? M.NumRows() : M.NumCols()));
```

Comment:
thanks.  Would you mind adding the check, in this branch, that the dim is correct?  Like in my PR?

Reply:
cancel that, I see that you did that.

Reply:
The line above your comment is checking 'dim_'. Do you mean something else?